core/reactor.h

```diff
@@ -548,6 +548,30 @@ struct trigger_t {
                                         // Note: The physical_time_of_arrival is only passed down one level of the hierarchy. Default: NEVER.
 #endif
 };
+
+/**
+ * An allocation record that is used by a destructor for a reactor
```

Comment:
I think this allocation record might have a lot of potential beyond just reactors! Perhaps people working on bare-iron might find this abstraction very useful for other parts of reactor-c.